src/main.rs

```diff
@@ -711,41 +708,77 @@ impl Context {
-        let internals_url = discourse.create_topic(
-            announcements_category,
-            &format!("Rust {} pre-release testing", version),
-            &internals_contents,
-        )?;
-        let blog_contents = if let Some(contents) =
-            self.config
-                .blog_contents(version, &self.date, true, Some(&internals_url))
-        {
-            contents
-        } else {
-            eprintln!("Skipping blogging - insufficient information to create blog post");
-            return Ok(());
-        };
-
-        let mut token = github.token(repository_for_blog)?;
-        token.create_file(
-            BLOG_PRIMARY_BRANCH,
-            &format!(
-                "posts/inside-rust/{}-{}-prerelease.md",
-                chrono::Utc::today().format("%Y-%m-%d"),
+            let announcements_category = 18;
+            let internals_url = discourse.create_topic(
+                announcements_category,
+                &format!("Rust {} pre-release testing", version),
+                &internals_contents,
+            )?;
+            let blog_contents = if let Some(contents) = self.config.stable_dev_static_blog_contents(
                 version,
-            ),
-            &blog_contents,
-        )?;
+                &self.date,
+                true,
+                Some(&internals_url),
+            ) {
+                contents
+            } else {
+                eprintln!("Skipping blogging - insufficient information to create blog post");
+                return Ok(());
+            };
+
+            let mut token = github.token(repository_for_blog)?;
+            token.create_file(
+                BLOG_PRIMARY_BRANCH,
+                &format!(
+                    "posts/inside-rust/{}-{}-prerelease.md",
+                    chrono::Utc::today().format("%Y-%m-%d"),
+                    version,
+                ),
+                &blog_contents,
+            )?;
+        } else if let Some(pr) = self.config.blog_pr {
+            // Otherwise, if we're passed a blog PR number, then we will try to merge that PR.
+            //
+            // We also post to Discourse with a release announcement once the PR is merged.
+            let version = self.current_version.as_ref().expect("has current version");
+            let mut token = github.token(repository_for_blog)?;
+            if let Err(e) = token.merge_pr(pr) {
+                eprintln!("Failed to merge PR: {:?}", e);
+                return Ok(());
+            }
+
+            // This is a pretty poor way of doing this but it doesn't seem obvious that there are
+            // massively better alternatives. GitHub pages also has a bug around clicking on links
+            // too soon causing the failure to find the artifact to get cached, resulting in longer
+            // publication times.
+            println!("Sleeping for 2 minutes to let merged PR build and deploy.");
+            std::thread::sleep(Duration::from_secs(120));
```

Comment:
I'm not happy about this but I agree there isn't a *simple* way to wait for the blog to be deployed. I guess we could poll for a new commit being pushed to the `gh-pages` branch, then poll the deployments api to wait for github pages to be deployed for that commit, but it's a bit overkill :(

Reply:
Yeah, I think if we wanted to do this right I'd probably suggest moving/duplicating the posting to internals/users to triagebot or something which is receiving webhook here.

Reply:
Thinking about it, a solution would be to move the blog deployment to be GitHub Actions-based, and add fields like this to the blog frontmatter:

```yaml
cross-post:
  urlo: true
  twitter: Rust 1.67.0 has been released! Check out the hightlights on our blog.
```

Then GitHub Actions could run a script after the files are published to do the cross-posting. That way we could use it for other posts in addition to release ones.

Reply:
Actually, a simpler way to implement this in `promote-release` would be to list the deployments before merging the blog, merge the blog, and then poll for a new completed deployment to appear on the list. That would be fairly simple to implement.

Reply:
I'm not sure that the end of deployment perfectly lines up with availability of content.

Is my comment about the GitHub pages bug still accurate? If not, then we could implement polling on the blog URL, which is simpler than the several APIs we'd otherwise be adding (though that's also possible of course).

Reply:
@pietroalbini I'm feeling inclined to merge this as-is. The polling you suggest does seem relatively straightforward to add but it also feels iffy on being strictly sufficient (per my previous comment, I'm not convinced pages deployment and content availability are 1:1 aligned).

It also seems like the added complexity of polling is higher than just assuming that things are going to be done in 2 minutes (which feels like a reasonable target), I'm not convinced that's worth it.

Reply:
@Mark-Simulacrum I think the content availability should be fixed when the deploy becomes green. I can't seem to find it on most repositories, but at work I have a workflow deploying a large GitHub Pages website (~700MB) and there one of the steps the build waits for is "synchronizing content".

If possible I'd still add polling for the deployment.

Reply:
Ok, sounds good. I'll go ahead and implement that soon and poke for another review.

I think long-run we should probably consider having promote-release emit an event (e.g., to eventbridge, sns, or just some custom https url space) so we can handle this kind of thing in triagebot or elsewhere that's already setup for hooks and such. But it will add a lot of complexity so I think not worth doing just yet.

Reply:
I would like to eventually migrate this directly into the blog.rust-lang.org CI, as a step that runs after pages finished deploying. Posting on the users forum and ideally twitter too should be something to do for most of our blog posts, and the post/tweet content being just yet another item in the frontmatter would be so convenient.